src/cpp/cpp_typecheck_code.cpp

```diff
@@ -191,8 +198,10 @@ void cpp_typecheckt::typecheck_switch(codet &code)
     assert(decl.operands().size()==1);
 
     // replace declaration by its symbol
-    assert(decl.op0().op0().id()==ID_symbol);
-    value = decl.op0().op0();
+    DATA_INVARIANT(
+      to_unary_expr(decl).op().op0().id() == ID_symbol,
```

Comment:
How do we know that the `.op0()` is safe here?

Reply:
added a check